### PR TITLE
`spin add`

### PR DESCRIPTION
**Pull request by itowlson**
Signed-off-by: itowlson <ivan.towlson@fermyon.com>

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [ ] The `title, `template`, and `date` are all set
- [ ] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [ ] Has passed [`bart check`](https://bartholomew.fermyon.dev/quickstart)
- [ ] Has run `npm run build-index`
- [ ] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [ ] Headings are using Title Case
- [ ] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [ ] Have tested with `npm run test` and resolved all errors

### CODE REVIEW

**Comment by radu-matei** [comment]
This looks great!
Do we want to add a reference in the document about developing applications (https://developer.fermyon.com/spin/developing) about the existence of this command?

I don't have a strong preference either way, so I am going to LGTM this.

Thank you!

**Review by radu-matei — APPROVED** [review]

**Comment by karthik2804** [comment]
Should we add a note that it is currently available only on `canary` and not on `v0.6.0`?

**Comment by mikkelhegn** [comment]
@tpmccallum - is the CLI reference still automated? E.g., will this change be overwritten eventually?

**Comment by radu-matei** [comment]
> Should we add a note that it is currently available only on canary and not on v0.6.0?

@karthik2804 — the doc updates are in preparation of v0.7.

**Comment by itowlson** [comment]
@radu-matei I looked at the "Developing Spin Applications" but at present it is entirely about the `build` command and configuration.  We should rework it, possibly into multiple pages, to cover the full create-build-test development loop, but I'm calling that as out of scope for this PR.

File: content/cloud/cli-reference.md [hunk 1]
```diff
@@ -264,6 +265,8 @@ OPTIONS:
                                        specified in the file
 ```
 
+The `spin add` command is identical to `spin new` except that it adds a component to an existing application (instead of starting a new application).
```
[the remark this answers is not in the document]
Comment (itowlson) [reply content/cloud/cli-reference.md]: Good call, done.

**Pull request merged**